js/wazirx.js

```diff
@@ -0,0 +1,710 @@
+const Precise = require ('./base/Precise');
+
+module.exports = class wazirx extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'wazirx',
+            'name': 'WazirX',
+            'countries': [ 'IN' ],
+            'version': 'v2',
+            'rateLimit': 100,
+            'has': {
+                'cancelAllOrders': true,
+                'cancelOrder': true,
+                'CORS': false,
+                'createOrder': true,
+                'fetchCurrencies': false,
+                'fetchBalance': true,
+                'fetchBidsAsks': false,
+                'fetchClosedOrders': false,
+                'fetchDepositAddress': false,
+                'fetchDeposits': true,
+                'fetchFundingFees': false,
+                'fetchFundingHistory': false,
+                'fetchFundingRate': false,
+                'fetchFundingRates': false,
+                'fetchMarkets': true,
+                'fetchMyTrades': false,
+                'fetchOHLCV': false,
+                'fetchOpenOrders': true,
+                'fetchOrder': true,
+                'fetchOrders': true,
+                'fetchOrderBook': true,
+                'fetchPositions': false,
+                'fetchStatus': false,
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchTime': true,
+                'fetchTrades': false,
+                'fetchTradingFee': false,
+                'fetchTradingFees': false,
+                'fetchTransactions': false,
+                'fetchWithdrawals': false,
+                'setLeverage': false,
+                'withdraw': false,
+                'fetchDepositAddressesByNetwork': false,
+                'transfer': false,
+                'fetchTransfers': false,
+            },
+            'urls': {
+                'logo': 'https://user-images.githubusercontent.com/1294454/148647666-c109c20b-f8ac-472f-91c3-5f658cb90f49.jpeg',
+                'api': 'https://api.wazirx.com/sapi/v1',
+                'www': 'https://wazirx.com',
+                'doc': 'https://docs.wazirx.com/#public-rest-api-for-wazirx',
+                'fees': 'https://wazirx.com/fees',
+            },
+            'api': {
+                'public': {
+                    'get': {
+                        'exchangeInfo': 1,
+                        'depth': 1,
+                        'ping': 1,
+                        'systemStatus': 1,
+                        'tickers/24hr': 1,
+                        'ticker/24hr': 1,
+                        'time': 1,
+                        'trades': 1,
+                    },
+                },
+                'private': {
+                    'get': {
+                        'account': 1,
+                        'allOrders': 1,
+                        'funds': 1,
+                        'historicalTrades': 1,
```

Comment:
Looks like we're missing an implementation of `fetchMyTrades` with this endpoint.

Reply:
@kroitor 
I don't think that's the case because this is a public endpoint.

Wazirx has 2 public endpoints to fetchTrades:
 -  (old trades) https://docs.wazirx.com/#old-trade-lookup-market_data
 - (recentTrades) https://docs.wazirx.com/#recent-trades-list

I've implemented fetchTrades method using the latter although I'm not sure how to handle this situation where we have two endpoints that overlap a bit. 

What do you think?

Reply:
@kroitor Although it's private, returns all trades 
![image](https://user-images.githubusercontent.com/43336371/148814087-20d4bff8-e68d-46c0-aab7-b09adce7b871.png)

Reply:
Ok, then we're fine. We could make this endpoint optional in fetchTrades then, like we do here, for example:

- https://github.com/ccxt/ccxt/blob/master/js/okex.js#L583
- https://github.com/ccxt/ccxt/blob/master/js/okex.js#L2250

Or here:

- https://github.com/ccxt/ccxt/blob/master/js/binance.js#L730
- https://github.com/ccxt/ccxt/blob/master/js/binance.js#L2342

Reply:
@kroitor added, followed binance approach 🙂